os/arch/arm/src/imxrt/imxrt_qtmr.h

```diff
@@ -1,440 +1,462 @@
+	IMXRT_QTMR_CH_MAX
+};
+
+typedef enum _qtmr_selection {
+	kQTMR_0,
+	kQTMR_1,
+	kQTMR_2,
+	kQTMR_3,
+	kQTMR_MAX,
+} qtmr_selection_t;
+
+/*!
+ * @brief Quad Timer config structure
+ *
+ * This structure holds the configuration settings for the Quad Timer peripheral. To initialize this
+ * structure to reasonable defaults, call the QTMR_GetDefaultConfig() function and pass a
+ * pointer to your config structure instance.
+ *
+ * The config struct can be made const so it resides in flash
+ */
+typedef struct _qtmr_config {
+	qtmr_primary_count_source_t primarySource; /*!< Specify the primary count source */
+	qtmr_input_source_t secondarySource;       /*!< Specify the secondary count source */
+	bool enableMasterMode;                     /*!< true: Broadcast compare function output to other counters;
+													false no broadcast */
+	bool enableExternalForce;                  /*!< true: Compare from another counter force state of OFLAG signal
+													false: OFLAG controlled by local counter */
+	uint8_t faultFilterCount;                  /*!< Fault filter count */
+	uint8_t faultFilterPeriod;                 /*!< Fault filter period;value of 0 will bypass the filter */
+	qtmr_debug_action_t debugMode;             /*!< Operation in Debug mode */
+} qtmr_config_t;
+
+struct imxrt_qtmr_chipinfo_s {
+	TMR_Type   *base;
+	int         irq_id;
+};
+
+
+/*******************************************************************************
+ * API
+ ******************************************************************************/
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+/*!
+ * @name Initialization and deinitialization
+ * @{
+ */
+
+/*!
+ * @brief Ungates the Quad Timer clock and configures the peripheral for basic operation.
+ *
+ * @note This API should be called at the beginning of the application using the Quad Timer driver.
+ *
+ * @param base     Quad Timer peripheral base address
+ * @param channel  Quad Timer channel number
+ * @param config   Pointer to user's Quad Timer config structure
+ */
+void imxrt_qtmr_init(TMR_Type *base, qtmr_channel_selection_t channel, const qtmr_config_t *config);
+
+/*!
+ * @brief Stops the counter and gates the Quad Timer clock
+ *
+ * @param base     Quad Timer peripheral base address
+ * @param channel  Quad Timer channel number
+ */
+void imxrt_qtmr_deinit(TMR_Type *base, qtmr_channel_selection_t channel);
+
+/*!
+ * @brief  Fill in the Quad Timer config struct with the default settings
+ *
+ * The default values are:
+ * @code
+ *    config->debugMode = kQTMR_RunNormalInDebug;
+ *    config->enableExternalForce = false;
+ *    config->enableMasterMode = false;
+ *    config->faultFilterCount = 0;
+ *    config->faultFilterPeriod = 0;
+ *    config->primarySource = kQTMR_ClockDivide_2;
+ *    config->secondarySource = kQTMR_Counter0InputPin;
+ * @endcode
+ * @param config Pointer to user's Quad Timer config structure.
+ */
+void imxrt_qtmr_getdefaultconfig(qtmr_config_t *config);
+
+/*! @}*/
```

Comment:
What is the use of this line?

Reply:
Sorry, I didn't get you. What are you referring to ?

Reply:
/*! @}*/ ==>  Do we need this comment? What exactly it does? In TizenRT we have different commenting style so

Reply:
@drashti304 , It's not a comment. It is the pattern followed by the nxp. I prefer to keep it the same in this PR. May be we could modify it in the next PR.

Reply:
Alright